app/preview.py

```diff
@@ -65,6 +65,27 @@ def page_count():
     )
 
 
+@statsd(namespace="template_preview")
+def get_pdf_redis_key(json):
+    def print_dict(d):
+        """
+        From any environment/system will return the same string representation of a given dict.
+
+        ASSUMPTION: all dict values are immutable (eg int, none, str)
```

Comment:
This assumption is incorrect. Personalisation values can be lists.

Reply:
ahhhh nuts 😓

Reply:
Just needs a slightly smarter algorithm here to flatten everything into something hashable. Not a massive change I don’t think.

Reply:
so the implementation's changed (we tried using `frozenset`) and the comment's out of date - immutability is no longer a requirement, but sortability is - you can't sort a list of disparate types, so i think we'll just cast everything to a string before sorting.

Reply:
want to avoid hashing, at least to start, so we can read the redis keys for debugging.

Reply:
we've added a test that makes sure it does sort mixed value types (`str`, `int`, `None`, `bool`, and `list`)